item.rb

```diff
@@ -0,0 +1,42 @@
+class Item
+  attr_reader :archived
+  attr_accessor :id, :genre, :author, :source, :label, :publish_date
+
+  def initialize(publish_date, archived: false)
+    @id = Random.rand(1..100)
+    @genre = []
+    @author = []
+    @source = []
+    @label = []
+    @publish_date = publish_date
+    @archived = archived
+  end
+
+  def add_genre(genre)
+    @genre << genre
+  end
+
+  def add_author(author)
+    @author << author
+  end
+
+  def add_source(source)
+    @source << source
+  end
+
+  def add_label(label)
+    @label << label
+  end
```

Comment:
- [ ] I think you can improve on these methods by adding a condition that prevents addition of a new author for example when the same author already exists. it is simple. Let  me show you the hint below 👇 
```ruby
 def genre=(genre)
      @genre = genre
      genre.items.push(self) unless genre.items.include?(self)
    end

    def author=(author)
      @author = author
      author.items.push(self) unless author.items.include?(self)
    end

    def label=(label)
      @label = label
      label.items.push(self) unless label.items.include?(self)
    end
```

Can you notice the unless statement?
Please update your methods too 😃 👍